lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLUtils.java

```diff
@@ -1019,9 +1017,16 @@ public static String createCardinalitySql(List<DatabaseColumnInfo> list, IdAndVe
 			if(!isFirst){
 				builder.append(", ");
 			}
-			builder.append("COUNT(DISTINCT ");
-			builder.append(info.getColumnName());
-			builder.append(") AS ");
+			// There is no need to run the actual count for columns for which an index is not created 
+			// or for metadata columns (such as row id) that manage their own indices
+			if (info.isMetadata() || !info.getType().isCreateIndex()) {
+				builder.append(TableConstants.COLUMN_NO_CARDINALITY);

```

Comment:
if we use max(1) then we always get one row back even for an empty table.  This should be super cheap.